src/app/components/mnoe-users-local-list/mnoe-users-local-list.html

```diff
@@ -40,8 +40,13 @@
             <span ng-show="user.status === 'invited'" ng-click="removeUserFromOrganization(user)" uib-tooltip="{{'mnoe_admin_panel.dashboard.users.widget.local_list.cancel_invite' | translate}}"><i class="fa fa-close fa-1"></i></span>
           </td>
           <td ng-if="organizationContext">
-            <select ng-model="user.role" ng-options="role as keyFromRole(role) | translate for role in availableRoles" ng-change="updateUserRole(user, '{{user.role}}')"></select>
+            <select ng-show="user.editMode" ng-model="user.role" ng-options="role as usersRole.keyFromRole(role) | translate for role in usersRole.availableRoles"></select>
             <span ng-show="user.isUpdatingRole"><i class="fa fa-spinner fa-pulse fa-fw"></i></span>
+            <span ng-click="user.editMode = true" ng-hide="user.editMode">{{ usersRole.keyFromRole(user.role) | translate }}</span>
+            <a class="role_edit_link" ng-hide="user.editMode" ng-click="user.editMode = true"><i class="fa fa-pencil"></i></a>
+            <a class="role_edit_link" ng-show="user.editMode" ng-click="updateUserRole(organization, user, user.role, availableRoles)"><i class="fa fa-check"></i></a>
+            <a class="role_edit_link" ng-show="user.editMode" ng-click="user.editMode = false"><i class="fa fa-times"></i></a>
```

Comment:
If you edit the role without saving and click cancel, I think the new role would still be displayed in the UI?

Reply:
Sure does, will find fix for this.